src/VitessceGrid.js

```diff
@@ -41,101 +40,100 @@ function shallowEqual(objA, objB) {
-  }
-
-  render() {
-    const {
-      layout, getComponent, padding, margin, draggableHandle,
-      reactGridLayoutProps, onAllReady, rowHeight,
-    } = this.props;
-
-    const ResponsiveGridLayout = WidthProvider(Responsive);
-
-    const {
-      cols, layouts, breakpoints, components,
-    } = resolveLayout(layout);
-
-    // Inline CSS is generally avoided, but this saves the end-user a little work,
-    // and prevents class names from getting out of sync.
-    const style = (
-      <style>
-        {`
+const VitessceGridComponent = ({
+  k, v, Component, onReady, removeGridComponent,
+}) => (
+  <div key={k}>
+    <Component
+      {... v.props}
+      clearComponent={removeGridComponent}
+      onReady={onReady}
+    />
+  </div>
+);
+
+const VitessceGrid = (props) => {
+  const {
+    layout, getComponent, padding, margin, draggableHandle,
+    reactGridLayoutProps, onAllReady, rowHeight,
+  } = props;
+  const ResponsiveGridLayout = WidthProvider(Responsive);
+  const {
+    cols, layouts, breakpoints, components,
+  } = resolveLayout(layout);
+
+  // eslint-disable-next-line no-unused-vars
+  const [readyComponentKeys, setReadyComponentKeys] = useState(new Set());
+  const [gridComponents, setGridComponents] = useState(components);
+
+  // Inline CSS is generally avoided, but this saves the end-user a little work,
+  // and prevents class names from getting out of sync.
+  const style = (
+    <style>
+      {`
           ${draggableHandle} {
             cursor: grab;
           }
           ${draggableHandle}:active {
             cursor: grabbing;
           }
-        `}
-      </style>
-    );
-
-    const layoutChildren = Object.entries(components).map(([k, v]) => {
-      const Component = getComponent(v.component);
-      const onReady = () => {
-        this.setState((state) => {
-          const { readyComponentKeys } = state;
-          const newReadyComponentKeys = new Set(readyComponentKeys);
-          newReadyComponentKeys.add(k);
-          if (newReadyComponentKeys.size === Object.keys(components).length) {
-            // The sets are now equal.
-            onAllReady();
-          }
-          return { readyComponentKeys: newReadyComponentKeys };
-        });
-      };
-      return (
-        <div key={k}>
-          <Component {... v.props} onReady={onReady} />
-        </div>
-      );
+          `}
+    </style>
+  );
+
+  const layoutChildren = Object.entries(gridComponents).map(([k, v]) => {
+    const Component = getComponent(v.component);
+    const onReady = () => {
+      setReadyComponentKeys((prevReadyComponentKeys) => {
+        prevReadyComponentKeys.add(k);
+        if (prevReadyComponentKeys.size === Object.keys(gridComponents).length) {
+          // The sets are now equal
+          onAllReady();
+        }
+        return prevReadyComponentKeys;
+      });
+    };
+
+    const removeGridComponent = () => {
+      const { [k]: _, ...newGridComponents } = gridComponents;
```

Comment:
I wouldn't have thought of this! ... Though I worry a bit about readability vs something like a plain `delete`.

Reply:
Agreed. I was trying to find a way to isolate a copy of the elements we want to update state to, and utilized dynamic destructing for this. Probably a bit too hacky. 

Changed to:

```javascript
    const removeGridComponent = () => {
      const newGridComponents = { ...gridComponents };
      delete newGridComponents[k];
      setGridComponents(newGridComponents);
    };
```